RummikubGameApp/RummikubGameApp/GameLogic.cs

```diff
@@ -0,0 +1,340 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Drawing;
+using System.Windows.Forms;
+using System.Text.RegularExpressions;
+
+namespace RummikubGameApp
+{
+    public class GameLogic

```

Comment:
your code conventions are all over the place, from [checkdeck(all small), to CheckLegalDeck (capital on word) or this.AIhand (no capital h) or just hand  (no this) or this.currentPlayerIndex. please pick one for each case: public method, private method, variables